### ISSUE TITLE
Exception thrown might be misleading

### ISSUE THREAD

**Issue by rvazquezglez** [jsalazare/just-crud#2]
What about returning an Optional instead of throwing an exception. In that way, the bean using the service will be forced to handle the case where the user is not found instead of failing with an exception.

https://github.com/jsalazare/just-crud/blob/6319a77529cb346e242b4c8952f2eca32089c200/src/main/groovy/com/jsalazar/justcrud/service/UserServiceImpl.groovy#L25

**Comment by jsalazare** [comment]
Yes i can move the exception handling to the controller, i just wanted to make sure that every time we get a user from `userRepository` this one is not null. Returning an optional will make us to handle this exception on every controller or service i want to get a `userById()`.

I think this is more like a RuntimeException, i just upload this change.
